src/box/box.cc

```diff
@@ -1453,6 +1453,8 @@ box_check_replication_synchro_quorum(void)
 static double
 box_check_replication_synchro_timeout(void)
 {
+	if (!replication_synchro_timeout_enabled)
+		return TIMEOUT_INFINITY;
```

Comment:
Please support `box.begin{timeout=...}`. If a quorum can't be gathered the user should receive an error and not be blocked indefinitely, but the transaction should remain in limbo.

Now if I try this on a single instance:
```lua
tarantool> box.cfg{replication_synchro_quorum=2}
2024-04-03 16:16:44.714 [58149] main/104/interactive I> RAFT: fencing paused
2024-04-03 16:16:44.714 [58149] main/104/interactive/box.load_cfg I> set 'replication_synchro_quorum' configuration option to 2
---
...

tarantool> box.begin({timeout=0.01}) box.space.sync:replace{1} box.commit()


```
the request hangs.

Reply:
IIRC we agreed to open a new issue with this request: one should be able to set commit timeout for a transaction. After a timeout passes, commit() returns an error, but the synchronous transaction isn't rolled back.

Reply:
Ticket: https://github.com/tarantool/tarantool/issues/10181

Reply:
Thanks!